katharsis-core/src/main/java/io/katharsis/utils/parser/TypeParser.java

```diff
@@ -105,7 +106,11 @@ public TypeParser() {
 		} else {
 			Method method;
 			try {
-				method = clazz.getMethod("parse", String.class);
+				try {
```

Comment:
This whole `parse` method is god awful horrible! This whole function can and should be reduced to a `Map<Class<T>,Function<String,T>` and then statically map them. Or, and in addition to, you can make a Map of 
```
		Map<Predicate<Class<? extends T>>,Function<String,T>> supportedTypes = new HashMap<>();
		supportedTypes.put(c -> c.isInstance(Enum.class), s -> Enum.valueOf((Class<Enum>) clazz.asSubclass(Enum.class), input.trim()));
//...
		supportedTypes.entrySet().stream().filter(e -> e.getKey().test(clazz)).findFirst().map(Map.Entry::getValue).map(f -> f.apply(input)).orElseThrow(IllegalArgumentException::new);

```